gmsh_utils/math_utils.py

```diff
@@ -0,0 +1,177 @@
+from typing import Sequence
+
+import numpy as np
+import numpy.typing as npt
+
+
+class MathUtils:
+    """
+    Class with mathematical utilities.
+    """
+
+    @staticmethod
+    def calculate_rotation_matrix_polygon(polygon_vertices: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
+        """
+        Calculate the rotation matrix to align the polygon with the x-y plane. Using the Rodrigues' rotation formula.
+
+        Args:
+            - polygon_vertices (npt.NDArray[np.float64]): Vertices of the polygon.
+
+        Returns:
+            - npt.NDArray[np.float64]: Rotation matrix.
+        """
+
+        polygon_normal = MathUtils.calculate_normal_plane(polygon_vertices)
+
+        # Define the target normal (Z-axis in this case)
+        target_normal = np.array([0, 0, 1])
+
+        # Calculate the rotation vector
+        rotation_vector = np.cross(polygon_normal, target_normal)
+
+        # Calculate sin and cos of the angle between the polygon normal and the target normal
+        sin_theta = np.linalg.norm(rotation_vector)
+        cos_theta = np.dot(polygon_normal, target_normal)
+
+        # Normalize the rotation vector
```

Comment:
normalisation is rotation_vector = rotation_vector / np.linalg.norm(rotation_vector).

Reply:
non normalised rotation vector = np.cross(polygon_normal, target_normal),

||a × b|| = ||a|| * ||b|| * sin(θ)

||np.cross(polygon_normal, target_normal)|| = ||polygon_normal|| * ||target_normal|| * sin(θ)
If your normals are unit vectors (which they typically should be for surface normals), then ||polygon_normal|| = ||target_normal|| = 1, so it simplifies to:
||np.cross(polygon_normal, target_normal)|| = sin(θ)

in short: sin(θ) == np.linalg.norm(non_normalised_rotation_vector)